src/arm/debugger/debugger.c

```diff
@@ -126,6 +126,10 @@ bool ARMDebuggerSetSoftwareBreakpoint(struct mDebuggerPlatform* d, uint32_t addr
 		return false;
 	}
 
+	if (address >= 0x3FFF) {
```

Comment:
That's GBA specific, but the ARM core is supposed to be generic. This shouldn't be done here. There isn't really a good way to do this, though, without changing a bunch of stuff to pipe it up into the GBA-specific code.